src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs

```diff
@@ -224,10 +230,19 @@ void InvokeTryProcessResponse(Task responseTask)
                     {
                         return;
                     }
+
+                    if (responseTask.Status == TaskStatus.Faulted)
+                    {
+                        transaction.NoticeError(responseTask.Exception);
+                    }
+
                     var responseGetter = _getRequestResponseFromGeneric ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<object>(responseTask.GetType(), "Result");
                     var response = responseGetter(responseTask);
                     if (_functionDetails.EventType is AwsLambdaEventType.APIGatewayProxyRequest or AwsLambdaEventType.ApplicationLoadBalancerRequest)
                         CaptureResponseData(transaction, response);
+
+                    segment.End();
+                    transaction.End();
```

Comment:
This should probably also happen in the error case too, before the early return.

Reply:
Maybe put it in a `finally` block?